hack/lib/golang.sh

```diff
@@ -161,15 +161,31 @@ readonly KUBE_ALL_TARGETS=(
 )
 readonly KUBE_ALL_BINARIES=("${KUBE_ALL_TARGETS[@]##*/}")
 
+# These binaries does not contain any C code at all, and may therefore be compiled statically without involving gcc
+# Binaries that does contain C code are also compiled statically, but there gcc first statically compiles the C code, and the
+# statically linked C code is then injected into static Go code.
+# Again, this lists all those binaries that do not contain C code:
```

Comment:
I know it is backwards, instead of listing those binaries that need CGO, we're listing the binaries that don't. Right now the "default" is to compile with CGO, although the default should be the opposite.

I may take a look how to clean it up later